tests/baselines/reference/redefineArray.types

```diff
@@ -1,8 +1,8 @@
 === tests/cases/compiler/redefineArray.ts ===
 Array = function (n:number, s:string) {return n;};
->Array = function (n:number, s:string) {return n;} : (n: number, s: string) => number
+>Array = function (n:number, s:string) {return n;} : <T>(n: number, s: string) => number

```

Comment:
`Array`'s call signature has type parameters (and multiple overloads, hence why we only now pick it up), so our contextual signature logic says to persist those type parameters onto the inferred signature (so they can be inferred and used at the parameter types). That type parameter just happens to end up unused, since the parameters are all annotated with types that override the inferred ones. Specifically, it has the signatures:
```
    (arrayLength?: number): any[];
    <T>(arrayLength: number): T[];
    <T>(...items: T[]): T[];
```
so it used to be that you'd get no contextual signature whatsoever, whereas now the contextual signature is something like `<T>(arrayLengthOrItem?: number | T, ...items: T[]): T[]`. Technically any time a contextually typed signature has all the parameter types specified, we could probably omit these type parameters. Probably. You can see similar behavior in the `types` baselines today with only one overload and something like:
```ts
// @strictFunctionTypes: false
interface MyCallable {
    <T>(a: T | number): T[];
}

const x: MyCallable = function (arg: number) { return null as any };
```
(`strictFunctionTypes` has to be off for the assignment to succeed, in both cases). I don't know _why_, but we also only do this for _function expressions_ and not _arrow functions_... that's probably indicative of some bug somewhere; but it's a preexisting one.

Reply:
OK, I figured it was pre-existing.